pkg/apis/common/common.go

```diff
@@ -103,24 +101,28 @@ type TLSConfig struct {
 
 	// DeprecatedCACertPath refers the file path that contains the CA cert.
 	// Deprecated: use CACertSecret instead
-	DeprecatedCACertPath string `json:"caCertPath" protobuf:"bytes,4,opt,name=caCertPath"`
+	DeprecatedCACertPath string `json:"caCertPath,omitempty" protobuf:"bytes,4,opt,name=caCertPath"`
 	// DeprecatedClientCertPath refers the file path that contains client cert.
 	// Deprecated: use ClientCertSecret instead
-	DeprecatedClientCertPath string `json:"clientCertPath" protobuf:"bytes,5,opt,name=clientCertPath"`
+	DeprecatedClientCertPath string `json:"clientCertPath,omitempty" protobuf:"bytes,5,opt,name=clientCertPath"`
 	// DeprecatedClientKeyPath refers the file path that contains client key.
 	// Deprecated: use ClientKeySecret instead
-	DeprecatedClientKeyPath string `json:"clientKeyPath" protobuf:"bytes,6,opt,name=clientKeyPath"`
+	DeprecatedClientKeyPath string `json:"clientKeyPath,omitempty" protobuf:"bytes,6,opt,name=clientKeyPath"`
 }
 
 // Backoff for an operation
 type Backoff struct {
-	// Duration is the duration in nanoseconds
-	Duration time.Duration `json:"duration" protobuf:"varint,1,opt,name=duration,casttype=time.Duration"`
+	// The initial duration in nanoseconds or strings like "1s", "3m"
```

Comment:
The goal of changing `Backoff` properties:

1. Make it easy to use, user can use strings like `2s`, `1m` for `Duration`;
2. Make it possible to only specify a property such as `steps`, and use default ones for the rest.